### PR TITLE
Changes to support both OneDNN and ZenDNN builds

### PR DESCRIPTION
**Pull request by kahmed10**
- updated CMakeLists.txt (might need some refactoring for cleanup)
- created macro to switch between OneDNN and ZenDNN API calls
- disabled some operators for lowering on ZenDNN since some unit tests currently fail
- moved deconvolution compute method to operator header file so it can fallback to ref implementation

### CODE REVIEW

**Review by pfultz2 — APPROVED** [review]

File: src/targets/cpu/dnnl.cpp [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 #if defined(__GNUC__) && __GNUC__ <= 5
 namespace std {
+#ifdef MIGRAPHX_ENABLE_ZENDNN
+namespace dnnl = zendnn;
```
Comment (umangyadav) [inline src/targets/cpu/dnnl.cpp:6]: Wouldn't `dnnl=zendnn` assignment come from `dnnl.hpp` ? If so, do we need another assignment here ?

File: src/targets/cpu/include/migraphx/cpu/dnnl.hpp [hunk 2]
```diff
@@ -7,14 +7,26 @@
 #include <migraphx/register_op.hpp>
 #include <migraphx/check_shapes.hpp>
 #include <unordered_map>
-#include <dnnl.hpp>
 #include <migraphx/errors.hpp>
 #include <migraphx/assert.hpp>
+#ifdef MIGRAPHX_ENABLE_ZENDNN
+#include <zendnn.hpp>
+#else
+#include <dnnl.hpp>
```
Comment (umangyadav) [inline src/targets/cpu/include/migraphx/cpu/dnnl.hpp:15]: `dnnl.hpp` including itself ? I guess not, can you add a comment here where it is coming from ?

Comment (kahmed10) [reply src/targets/cpu/include/migraphx/cpu/dnnl.hpp:15]: This file is `<migraphx/cpu/dnnl.hpp>`, and it's referencing the actual OneDNN and ZenDNN headers here.

File: src/targets/cpu/dnnl.cpp (hunk 1 above)
Comment (kahmed10) [reply src/targets/cpu/dnnl.cpp:6]: I am getting compiler errors with this removed.

Comment (umangyadav) [reply src/targets/cpu/dnnl.cpp:6]: If errors are not easily fixable then, it is okay, we can keep this.

Comment (pfultz2) [reply src/targets/cpu/dnnl.cpp:6]: dnnl.hpp does not do `dnnl=zendnn` as it is completely unaware of zendnn. Also zendnn does not provide such an alias either, so this is necessary.

**Review by umangyadav — APPROVED** [review]

**Pull request merged**